frontend/src/index.tsx

```diff
@@ -1424,6 +1424,21 @@ function renderUI() {
                 }}
               />
             </label>
+            {state.game_state.Initialize ? null : (
+              <a
+                href={window.location.href}
+                className="reset-link"
+                onClick={(evt) => {
+                  evt.preventDefault();
+                  if (window.confirm('Do you really want to reset the game?')) {
+                    send({Action: 'ResetGame'});
+                  }
+                  renderUI();
```

Comment:
A next step after this might be to make the entire application computable from the input state. So, for example:

- Create a top-level component that maintains the state
- Any time we receive a websocket input, use `setState` on it
- Here, fire-and-forget the `send`, and remove the imperative render

Reply:
yeah, that makes sense.

Most of the components are structured that way, but the things which directly render the state global are not (mostly; I didn't know how to)